### PR TITLE
Several changes to code base to make it easier to subclass the LBS so…

### PR DESCRIPTION
**Pull request by jthano**
…lver

### CODE REVIEW

**Comment by jthano** [comment]
The changes show that a new line was added to my modified files. Do the files in the code base not have a trailing new line to end the file? Maybe this is just an issue cause by my eclipse setup

File: Modules/LinearBoltzmanSolver/GroupSet/lbs_groupset.h [hunk 1]
```diff
@@ -72,6 +74,17 @@ class LBSGroupset
 
   double                                       latest_convergence_metric;
 
+  /**Convenient typdef for the moment call back function. See moment_callbacks*/
+  typedef std::function<void(int, int, int, double)> MomentCallbackF;
```
Comment (Naktakala) [inline Modules/LinearBoltzmanSolver/GroupSet/lbs_groupset.h]: The only part I am missing here is what the moment arguments are. We have two ways to address it. We could either include them into the typedef:

`typedef std::function<void(int dof, int moment, int angle_num, double psi)> MomentCallbackF;`

or we could include it in its doc-string.
`/**Convenient typdef for the moment call back function. See moment_callbacks.
Arguments are:
int dof_mapping Local DOF-address with moment and group folded in.
int moment         The moment number.
int angle_num    The reference angle number in the angular quadrature.
double psi          Angular flux for the given DOF.
*/`

In any case, we will make improvements later so just choose one.

**Review by Naktakala — CHANGES_REQUESTED** [review]
Just a commenting request. Just cause I can. Then approve

File: Modules/LinearBoltzmanSolver/GroupSet/lbs_groupset.h (hunk 1 above)
Comment (jthano) [reply Modules/LinearBoltzmanSolver/GroupSet/lbs_groupset.h]: I went ahead and updated the comment and added names to the argument list

**Pull request merged**